audit-service/src/component-tests/java/uk/gov/gchq/palisade/component/audit/model/AuditErrorMessageTest.java

```diff
@@ -16,88 +16,41 @@
 
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.json.JsonTest;
-import org.springframework.boot.test.json.JacksonTester;
-import org.springframework.boot.test.json.JsonContent;
-import org.springframework.boot.test.json.ObjectContent;
-import org.springframework.test.context.ContextConfiguration;
 
 import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.service.audit.AbstractSerialisationTest;
 import uk.gov.gchq.palisade.service.audit.model.AuditErrorMessage;
 
-import java.io.IOException;
-import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
-import java.util.HashMap;
 import java.util.Map;
 
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.jupiter.api.Assertions.assertAll;
+import static java.time.ZoneOffset.UTC;
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
 
-@JsonTest
-@ContextConfiguration(classes = {AuditErrorMessageTest.class})
-class AuditErrorMessageTest {
-
-    @Autowired
-    private JacksonTester<AuditErrorMessage> jsonTester;
+class AuditErrorMessageTest extends AbstractSerialisationTest {
 
     /**
-     * Grouped assertion test
      * Create the object with the builder and then convert to the Json equivalent.
      * Takes the JSON Object, deserialises and tests against the original Object
      *
-     * @throws IOException throws if the {@link AuditErrorMessage} object cannot be converted to a JsonContent.
-     *                     This equates to a failure to serialise or deserialise the string.
+     * @throws Exception if any error occurs during (de)serialisation
+     * @see AbstractSerialisationTest#testInstance(Object)
      */
     @Test
-    void testGroupedDependantErrorMessageSerialisingAndDeserialising() throws IOException {
-        Context context = new Context().purpose("testContext");
-        String now = ZonedDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
-        Map<String, Object> attributes = new HashMap<>();
-        attributes.put("messagesSent", "23");
+    void testAuditErrorMessageSerialisation() throws Exception {
 
-        AuditErrorMessage auditErrorMessage = AuditErrorMessage.Builder.create()
-                .withUserId("originalUserID")
-                .withResourceId("testResourceId")
-                .withContext(context)
-                .withServiceName("testServicename")
-                .withTimestamp(now)
-                .withServerIp("testServerIP")
-                .withServerHostname("testServerHostname")
-                .withAttributes(attributes)
-                .withError(new InternalError("Something went wrong!"));
+        var expected = AuditErrorMessage.Builder.create()
+            .withUserId("originalUserID")
+            .withResourceId("testResourceId")
+            .withContext(new Context().purpose("testContext"))
```

Comment:
More of a comment than anything else.  Here you have a Context object created for a test and again later on in another test.  We have a convention of dumping the test objects in a class that is shared by all of the tests for example in the component tests

Reply:
Single values are OK in test data classes such as (String, Date, int, double etc)/ But more complex objects should be created per test/test class.